Expose last deployed time via stack endpoint

lalithr95 opened the pull request:
Shipit api doesn't provide last deployment time through stack endpoint. This PR exposes last deployed time for each stack. This time would be useful in [services-db](https://github.com/Shopify/services-db/pull/871) to keep track of actively deployed services. Updated minor piece of unrelated code in stack mode and added `@stack.reload` in commit test since `@stack.commits.new` is going to instantiate new commit which doesn't have any foreign key data assigned.

@byroot @wvanbergen

casperisfine commented on test/dummy/db/schema.rb:
```diff
@@ -174,6 +174,7 @@
     t.datetime "continuous_delivery_delayed_since"
     t.datetime "locked_since"
     t.boolean  "merge_queue_enabled",                             default: false,        null: false
+    t.datetime "deployed_at"
```
I'd name it `last_deployed_at`.

casperisfine commented on app/models/shipit/deploy.rb:
```diff
@@ -180,5 +181,9 @@ def last_successful_deploy
     end
+
+    def update_last_deploy_time
+      stack.update(deployed_at: updated_at)
```
I'd use the `ended_at` timestamp.

lalithr95 replied:
Good one 👍

lalithr95 commented:
> Should we backfill the data?

No need to backfill column data, this way we could start with fresh data.
Fixed comments and added column to `StackSerializer`

casperisfine commented on test/models/deploys_test.rb:
```diff
@@ -225,6 +225,13 @@ def setup
+      @deploy = shipit_deploys(:shipit_running)
+      @deploy.complete!
+      @stack.reload
+      assert_in_delta @deploy.updated_at, @stack.last_deployed_at, 1
```
That test should be updated.
